gym_solo/core/rewards.py

```diff
@@ -60,4 +63,39 @@ def get_reward(self) -> float:
       combination of multiple atomic sub-rewards, as explained by the 
       strategies earlier.
     """
-    return sum(wr.weight * wr.reward.compute() for wr in self._rewards)
\ No newline at end of file
+    return sum(wr.weight * wr.reward.compute() for wr in self._rewards)
+
+
+class UprightReward(Reward):
+  """A reward for being fully upright. Note that this is technically only
+  designed for the Solo8v2Vanilla and should be extended in the future to 
+  support more models.
+
+  The reward's is in the interval [-1, 1], where 1 indicates that the torso
+  is upright, while -1 means that it is upside down. Observe that this means
+  that the reward is orientation-specific.
+  """
+  _fully_upright = -np.pi / 2
+
+  def __init__(self, robot_id: int):
+    """Create a new UprightReward.
+
+    Args:
+      robot_id ([int]): The PyBullet body id of the robot
+    """
+    self._robot_id = robot_id
+
+  def compute(self) -> float:
+    """Compute the UprightReward for the current state. 
+    
+    As this reward function is specifically designed for the solo8v2vanilla 
+    model, this means that the model just needs to be rotated -pi/2 radians 
+    about the y axis.
+
+    Returns:
+      float: A real-valued number in [-1, 1], where 1 means perfectly upright 
+      whilst -1 means that the robot is literally upside down. 
+    """
+    _, quat = p.getBasePositionAndOrientation(self._robot_id)
+    unused_x, y, unused_z = np.array(p.getEulerFromQuaternion(quat))
```

Comment:
Should we be consistent about using radians through all our code?
I don't think it's a good idea to have degrees someplace and radians in other places.
Radians are the standard in robotics

Reply:
I think we should force the radian standard. The only reason I keep the degrees functionality was in case we had gotten sensor that gave back values in degrees. By keeping the toggle, the dream was to simulate as much hardware as easily as possible.

With that being said, implementing rad->degrees is *literally* one LOC so that's a trivial feature. I'm ok with forcing radians for everywhere (and if needed, just make a two-second rad->degrees wrapper)

Reply:
Closing cause all of the calculations are in radians anyways

Reply:
That was my misunderstanding. You are using radians anyway